credential/auth.go

```diff
@@ -0,0 +1,78 @@
+package credential
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+	"github.com/piprate/json-gold/ld"
+)
+
+const ClaimToService = "toService"
+
+const ErrAuthClaim MessageError = "invalid auth claim"
+
+var _ Claim = (*AuthClaim)(nil)
+
+type AuthClaim struct {
+	ID        string
+	ToService string
+}
+
+func (ac *AuthClaim) From(vc *verifiable.Credential) error {
+	claims, ok := vc.Subject.([]verifiable.Subject)
+	if !ok {
+		return NewVCError(ErrMalformedSubject, nil)
+	}
+
+	if len(claims) != 1 {
+		return NewVCError(ErrExpectSingleClaim, nil)
+	}
+
+	toService, err := extractCustomStringClaim(&claims[0], ClaimToService)
+	if err != nil {
+		return NewVCError(ErrExtractClaim, err)
+	}
+
+	ac.ID = claims[0].ID
+	ac.ToService = toService
+
+	return nil
+}
+
+var _ Parser[*AuthClaim] = (*AuthParser)(nil)
+
+type AuthParser struct {
+	*credentialParser
+	ServiceID string
+}
+
+func NewAuthParser(serviceID string, documentLoader ld.DocumentLoader) *AuthParser {
+	return &AuthParser{
+		credentialParser: &credentialParser{documentLoader: documentLoader},
+		ServiceID:        serviceID,
+	}
+}
+
+func (ap *AuthParser) ParseSigned(raw []byte) (*AuthClaim, error) {
```

Comment:
Ok, I understand. It seems like a good approach to use and check the `proofPurpose` field. However, this will require some modifications to the `axoned` CLI, as it currently only supports signing VCs with the `assertionMethod` proof, which is the default value in the current library used to sign VCs. 

I suggest adding a flag to the `credential` command in the `axone` CLI to allow users to select the `proofPurpose` when signing a Verifiable Credential (VC). 

This also means that I will need to re-sign all my test data files. 😂

Reply:
Ah yes that's true we need to support that on the `axoned` as well!

Sorry for coming up a bit late with that suggestion 🙈

Reply:
Sounds good, let's do it that way :)